rasa/core/agent.py

```diff
@@ -392,156 +312,110 @@ class Agent:
+        self.model_id = model_metadata.model_id
 
         self._set_fingerprint(fingerprint)
 
         # update domain on all instances
-        self.tracker_store.domain = domain
+        self.tracker_store.domain = self.domain
         if hasattr(self.nlg, "responses"):
-            self.nlg.responses = domain.responses if domain else {}
+            self.nlg.responses = self.domain.responses if self.domain else {}
 
-        self.model_directory = model_directory
+        self.initialize_processor()
 
     @classmethod
     def load(
         cls,
         model_path: Union[Text, Path],
-        interpreter: Optional[NaturalLanguageInterpreter] = None,
         generator: Union[EndpointConfig, NaturalLanguageGenerator] = None,
         tracker_store: Optional[TrackerStore] = None,
         lock_store: Optional[LockStore] = None,
         action_endpoint: Optional[EndpointConfig] = None,
         model_server: Optional[EndpointConfig] = None,
         remote_storage: Optional[Text] = None,
-        path_to_model_archive: Optional[Text] = None,
-        new_config: Optional[Dict] = None,
-        finetuning_epoch_fraction: float = 1.0,
     ) -> "Agent":
         """Load a persisted model from the passed path."""
-        try:
-            if not model_path:
-                raise ModelNotFound("No path specified.")
-            if not os.path.exists(model_path):
-                raise ModelNotFound(f"No file or directory at '{model_path}'.")
-            if os.path.isfile(model_path):
-                model_path = get_model(str(model_path))
-        except ModelNotFound as e:
-            raise ModelNotFound(
-                f"You are trying to load a model from '{model_path}', "
-                f"which is not possible. \n"
-                f"The model path should be a 'tar.gz' file or a directory "
-                f"containing the various model files in the sub-directories "
-                f"'core' and 'nlu'. \n\n"
-                f"If you want to load training data instead of a model, use "
-                f"`agent.load_data(...)` instead. {e}"
-            )
-
-        core_model, nlu_model = get_model_subdirectories(model_path)
-
-        if not interpreter and nlu_model:
-            interpreter = rasa.core.interpreter.create_interpreter(nlu_model)
-
-        domain = None
-        ensemble = None
-
-        if core_model:
-            domain = Domain.load(os.path.join(core_model, DEFAULT_DOMAIN_PATH))
-            ensemble = (
-                PolicyEnsemble.load(
-                    core_model,
-                    new_config=new_config,
-                    finetuning_epoch_fraction=finetuning_epoch_fraction,
-                )
-                if core_model
-                else None
-            )
-
-            # ensures the domain hasn't changed between test and train
-            domain.compare_with_specification(core_model)
+        model_metadata, graph_runner = cls.unpack_model(model_path)
```

Comment:
do we need to catch potential exceptions here and convert them to `ModelNotFound`?